### PR TITLE
testing restaurantController w. reviews

### PR DESCRIPTION
**Pull request by isaac-fisher**

### CODE REVIEW

File: spec/factories/reviews.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :review do
+
```
Comment (liatperl) [inline spec/factories/reviews.rb:5]: the review should have a reviewer property or name property if you defined this property as mandatory in the review model

File: spec/factories/restaurants.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+FactoryBot.define do
+
+  # user factory without associated posts
+  factory :restaurant do
```
Comment (liatperl) [inline spec/factories/restaurants.rb:4]: Please try to write several tests that validate the model properties. For instance, check that the model is not valid if some of its mandatory properties are missing and that only when a value conform to a certain range the produced model is valid.

File: spec/factories/restaurants.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    association :cuisine, factory: :cuisine
+
+    factory :restaurant_with_reviews do
+      #ratings of reviews for this restaurant
+      transient do
+        ratings [1, 2]
+      end
+
+      after(:create) do |restaurant, evaluator|
+        evaluator.ratings.each{ |r| create(:review, restaurant: restaurant, rating: r)}
```
Comment (liatperl) [inline spec/factories/restaurants.rb:18]: great 👍

**Review by liatperl — CHANGES_REQUESTED** [review]

**Comment by isaac-fisher** [comment]
UPDATED @liatperl

File: spec/models/Restaurant_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+describe Restaurant do
+
+  it { should validate_presence_of(:name) }
+  it { should have_many(:reviews) }
+  it { should belong_to(:cuisine) }
+
+  it { should allow_values('SomeString').for(:address) }
+  it { should_not allow_values(Faker::Lovecraft.fhtagn(5)).for(:address) }
+
+  it { should allow_values('35').for(:delivery_time) }
```
Comment (liatperl) [inline spec/models/Restaurant_spec.rb]: Delivery time should be an integer in the range of 1 to 1440.'35' and '5000' doesn't meet this criteria, the test should have failed.

Comment (liatperl) [reply spec/models/Restaurant_spec.rb]: Maybe if you defined the delivery time as numericality: { only_integer: true } in the model the test would have failed, you can try that

File: spec/models/Restaurant_spec.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    end
+  end
+
+  context 'When has no reviews' do
+    let!(:restaurant) { FactoryBot.create :restaurant }
+    it 'Returns nil for ratings' do
+      expect(restaurant.rating).to be_nil
+    end
+  end
+
```
Comment (liatperl) [inline spec/models/Restaurant_spec.rb:30]: can you also test the associations in the model? I think this is a good practice

**Review by liatperl — APPROVED** [review]

**Pull request merged**